### PR TITLE
Add Tests For Controllers

### PR DESCRIPTION
**Pull request by peterdtitan**
## Summary
Following the project requirements outlined, I have done the following: 

- Added RSpec as a gem to the project.
- Created a Request spec file for all controllers.
- Made sure that for each action the following check were carried:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
```
Comment (Felix45) [inline spec/controllers/posts_controller_spec.rb:3]: - [ Required ] Good job adding feature tests for your controllers, however, it appears that some of the tests are failing when I run the `rspec spec/controllers/posts_controller_spec.rb` command on this file. Please see the screenshot below for more details on this. Also see this article on how to test controllers [How to test controllers >>](https://www.microverse.org/blog/understanding-rspec-controller-request-specs-integration-test-using-capybara)

## Error
Screenshot from 2023-06-21 23-55-22

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
```
Comment (Felix45) [inline spec/controllers/users_controller_spec.rb:3]: - [ Required ] Good job adding feature tests for your controllers, however, it appears that some of the tests are failing when I run the `rspec spec/controllers/users_controller_spec.rb` command on this file. Please see the screenshot below for more details on this. Also see this article on how to test controllers [How to test controllers >>](https://www.microverse.org/blog/understanding-rspec-controller-request-specs-integration-test-using-capybara)

## Error
Screenshot from 2023-06-21 23-58-42

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @peterdtitan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Application uses `rspec` gem for unit testing :white_check_mark:   
- [x] Repository has a professional readme file :white_check_mark:  
- [x] Pull request has a descriptive title and summary :white_check_mark: 


## Required Changes ♻️

- [Required] I think it would be better if you fixed this error in your `/app/views/layouts/application.html.erb` file, I believe it's one of the reasons why some of the controller tests are failing. I would like to suggest that you ensure the `tailwind` CSS library is imported correctly. You could try including the `tailwind` library by refactoring this line as follows because the name of the stylesheet is `application.tailwind.css` and not `tailwind.css`.

```ruby
  <%= stylesheet_link_tag "application.tailwind", "inter-font", "data-turbo-track": "reload" =%>
```
## Error
Screenshot from 2023-06-22 00-12-10



_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/controllers/users_controller_spec.rb (hunk 2 above)
Comment (peterdtitan) [reply spec/controllers/users_controller_spec.rb:3]: Done with passing tests. ✅ ✅ 

N.B: Tailwind css is correctly configured. Maybe missing installation from the cloned repo.

Screenshot 2023-06-22 at 08 58 27

**Comment by peterdtitan** [comment]
All tests passing as shown below: 

Screenshot 2023-06-22 at 09 14 02

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @peterdtitan , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Tailwind error fixed and app starts correctly
:heavy_check_mark: Tests all passing
:heavy_check_mark:  Good communication with reviewer

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**